share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorService.java

```diff
@@ -1090,4 +1111,10 @@ private void throwIfNotActive() {
             throw Errors.COORDINATOR_NOT_AVAILABLE.exception();
         }
     }
+
+    private boolean isShareGroupsEnabled(MetadataImage image) {
+        return ShareVersion.fromFeatureLevel(
+            image.features().finalizedVersions().getOrDefault(ShareVersion.FEATURE_NAME, (short) 0)
+        ).supportsShareGroups();
```

Comment:
We should have internal config.shareGroupConfig.shareGroupEnabled config check as well so we are on parity with other places.